openmls/src/group/managed_group/test_managed_group.rs

```diff
@@ -268,3 +273,88 @@ ctest_ciphersuites!(export_secret, test(ciphersuite_name: CiphersuiteName) {
                 .unwrap()
     )
 });
+
+#[test]
+fn test_invalid_plaintext() {
+    let ciphersuite_name = Ciphersuite::default().name();
+    println!("Testing ciphersuite {:?}", ciphersuite_name);
+    let ciphersuite = Config::ciphersuite(ciphersuite_name).unwrap();
+
+    // Some basic setup functions for the managed group.
+    let handshake_message_format = HandshakeMessageFormat::Plaintext;
+    let update_policy = UpdatePolicy::default();
+    let callbacks = ManagedGroupCallbacks::default();
+    let managed_group_config = ManagedGroupConfig::new(
+        handshake_message_format,
+        update_policy,
+        10,
+        0,
+        false,
+        callbacks,
+    );
+    let number_of_clients = 20;
+    let setup = ManagedTestSetup::new(
+        managed_group_config,
+        number_of_clients,
+        CodecUse::StructMessages,
+    );
+    // Create a basic group with more than 4 members to create a tree with intermediate nodes.
+    let group_id = setup.create_random_group(10, ciphersuite).unwrap();
+    let mut groups = setup.groups.borrow_mut();
+    let group = groups.get_mut(&group_id).unwrap();
+
+    let (_, client_id) = &group
+        .members
+        .iter()
+        .find(|(index, _)| index == &0)
+        .unwrap()
+        .clone();
+
+    let clients = setup.clients.borrow();
+    let client = clients.get(client_id).unwrap().borrow();
+
+    let (mls_message, _welcome_option) = client
```

Comment:
I'm not saying it's nicer. But something like this would allow to avoid the `drop`s below.
```suggestion
    let (mls_message, _welcome_option) = setup.clients.borrow().get(client_id).unwrap().borrow()
```